app/routes.py

```diff
@@ -110,26 +113,48 @@ def categoryHandler(id):
         except Exception as e:
             return str(e), 500
 
-
-    else: 
+    else:
         if (Categories.findById(id) is None):
             return 'fail', 204
         Categories.delete(id)
         return "success", 200
 
 ################################################################################
-#matches 
+# matches
 ################################################################################
-    
-@cross_origin()
-@app.route("/api/matches", methods = ["GET"])
-def getMatches():
-    app.logger.debug('This endpoint was hit!')
-    matches = Matches.query.all()
-    return json.dumps([m.serialize() for m in matches])
 
-@cross_origin()
-@app.route("/api/match/<id>", methods = ["DELETE", "GET", "PUT"])
+
+# @cross_origin()
+# @app.route("/api/matches", methods=["GET"])
+# def getMatches():
+#     # app.logger.debug('This endpoint was hit!')
+#     matches = Matches.query.all()
+#     return json.dumps([m.serialize() for m in matches])
+
+
+@cross_origin
+@app.route('/api/matches', methods=['GET'])
+def getMatchPage():
+    # app.logger.debug('hit match pagination')
+    page = request.args.get('page')
+    matchPerPage = 10
+    matches = Matches.query.all()
+    matches = matches[::-1]
+    matchPage = matches[(int(page)-1)*matchPerPage:int(page)*matchPerPage]
+    returnObj = {
+        'metadata':
+        {
+            'matchCount': str(int(len(matches))),
```

Comment:
lots of type casting here, is there a reason why we we need to cast `len(matches)` to an int?